docs/index.rst

```diff
@@ -1,49 +1,168 @@
-#######################
-Developer Documentation
-#######################
+.. Root index file for winter 2021 developer doc refresh. Content has been reorganized with an eye toward what the user wants to accomplish. 
+
+####################################
+Open edX Developer Documentation Hub
```

Comment:
I love this page compared to the legacy TOC. I still need to do a deeper review, but I think this is a million times nicer as a landing page.  I wish there was a more obvious way in the left sidebar to return to the TOC, but at least the landing page will be nice, and maybe someone will be able to refind it. :)